orchard/orchard_runner.go

```diff
@@ -8,28 +8,56 @@ package orchard
 import (
 	"encoding/json"
 	"fmt"
+	"log"
 	"os/exec"
+	"strings"
+
+	"mce.salesforce.com/sprinkler/common"
 )
 
 type OrchardRunner interface {
-	Generate(artifact string, command string) string
+	Generate(artifact string, command string) (string, error)
 }
 
 type OrchardStdoutRunner struct{}
 
-func (self OrchardStdoutRunner) Generate(artifact string, command string) (string, error) {
-	cmds, err := parseCommandLine(command)
-	if err != nil {
-		return "", err
-	}
-	if len(cmds) < 1 {
-		return "", fmt.Errorf("Invalid command line %s", command)
-	}
-	out, err := exec.Command(cmds[0], cmds[1:]...).Output()
-	if err != nil {
-		return "", err
+func (r OrchardStdoutRunner) Generate(artifact string, command string) (string, error) {
+	if !strings.HasPrefix(artifact, "s3://") {
```

Comment:
we still want to support the case where `artifact` is an empty string, which basically assumes the binary is already available in the sprinkler service instance.

Reply:
added logic to handle empty artifact